src/compass/utils/ale.py

```diff
@@ -0,0 +1,536 @@
+from osgeo import gdal
+from pyproj import CRS, Proj
+from shapely import geometry
+
+dateformat = '%Y-%m-%d %H:%M:%S.%f'
+
+
+def run(cslc_file, cr_file, csv_output_file=None, plot_age=False,
+        correct_set=False, mission_id='S1', pol='VV', ovs_factor=128,
+        margin=32):
+    '''
+    Compute Absolute Geolocation Error (AGE) for geocoded SLC
+    products from Sentinel-1 or NISAR missions. AGE is computed
+    by differencing the surveyed corner reflector (CR) positions from
+    the CSV file with that detected in the geocoded CSLC product.
+
+    Parameters
+    ----------
+    cslc_file: str
+        File path to HDF5 CSLC-S1 product
+    cr_file: str
+        File path to CSV file containing
+        surveyed CR locations
+    csv_output_file: str, None
+        File path to save computations in
+        CSV file. If None, csv file is not saved
+    plot_age: bool
+        If True, plots Absolute geolocation error
+        results
+    correct_set: bool
+        Correct Corner Reflector positions for Solid
+        Earth tides
+    mission_id: str
+        Mission identifier. 'S1' for Sentinel-1 OPERA
+        CSLC or 'NI' for NISAR GSLC
+    pol: str
+        Polarization channel to use to evaluate AGE
+    ovs_factor: int
+        Oversampling factor detecting CR location with
+        sub-pixel accuracy (default: 128)
+    margin: int
+        Margin to consider around CR position detected in
+        the geocoded SLC image. Overall included margin
+        is 2*margin from left-to-right and from top-to-bottom
+        (default: 32)
+    '''
+
+    # Check that the CSLC-S1 product file exists
+    if not os.path.exists(cslc_file):
+        err_str = f'{cslc_file} input geocoded SLC product does not exist'
+        raise FileNotFoundError(err_str)
+
+    # Check corner reflector file exists
+    if not os.path.exists(cr_file):
+        err_str = f'{cr_file} CSV CR position file does not exist'
+        raise FileNotFoundError(err_str)
+
+    # Open and load CSV CR file in pandas dataframe
+    cr_df = pd.read_csv(cr_file)
+
+    # Identify CRs contained in the usable part of the
+    # geocoded SLC
+    cslc_poly = get_cslc_polygon(cslc_file,
+                                 mission_id=mission_id)
+    # Initialize empty lists to include in the CR
+    # pandas dataframe
+    cr_x = []
+    cr_y = []
+    cr_x_cslc = []
+    cr_y_cslc = []
+
+    for idx, row in cr_df.iterrows():
+        # Extract surveyed CR positions from pandas dataframe
+        cr_lat = row['Latitude (deg)']
+        cr_lon = row['Longitude (deg)']
+        cr_loc = geometry.Point(cr_lon, cr_lat)
+
+        # If the CR is contained in the geocoded SLC product
+        # get its position in the SLC image, otherwise, drop
+        # the CR from the pandas dataframe
+        if cslc_poly.contains(cr_loc):
+            # Convert corner lat/lon coordinates in UTM
+            cslc_epsg = get_cslc_epsg(cslc_file, mission_id=mission_id)
```

Comment:
```suggestion
            cslc_epsg = get_cslc_epsg(cslc_file, mission_id=mission_id, pol=pol)
```

Reply:
Not sure why we need to specify the polarization here. We are reading the `epsg` from the `projection` dataset which should be common for every polarization